system/lib/emmalloc.c

```diff
@@ -374,17 +384,17 @@ static void dump_memory_regions()
     Region *r = (Region*)root;
     assert(debug_region_is_consistent(r));
     uint8_t *lastRegionEnd = root->endPtr;
-    MAIN_THREAD_ASYNC_EM_ASM(out('Region block '+ptrToString($0)+' - '+ptrToString($1)+ ' ('+($2>>>0)+' bytes):'),
+    MAIN_THREAD_ASYNC_EM_ASM(out('Region block '+ptrToString($0)+' - '+ptrToString($1)+ ' ('+toPtr($2)+' bytes):'),
```

Comment:
`toPtr` is defined using EM_JS, but that is meant to generate C functions? It does look like atm we also create a JS function of that exact name, so random JS can call it, but is that documented?

I would expect to see a function meant for JS use to be defined in an `EM_ASM`, though I admit that `EM_JS` does emit it at the toplevel automatically, which is nice. Perhaps we should document that behavior if we don't already?

Reply:
How would I do this with EM_ASM?  Can you put EM_ASM at top level?

Maybe we should doccument this behaviour?   But I also don't think its a problem to use this undocumented aspect here in out core libraries since if we ever change that it would be obvious and easy enough for us to update emmalloc (indeed it would not be possible for us to change it without breaking the emmalloc tests).

Reply:
With EM_ASM you can write to the global scope,
```js
EM_ASM({
  myFunc = () { console.log('hello world') };
});
..
EM_ASM({
  myFunc();
});
```

I agree this would not break silently, but it does feel odd to me to depend on this in this one unrelated test. Seems it could be weird to debug. Or do we already elsewhere use this?

Reply:
This doesn't seem like good match for 

```
EM_ASM({
  myFunc = () => { console.log('hello world') };
});
```

Because these functions get run a lot so I would then need to guard it somehow, or execute it ever time..